toolchain/check/check_unit.cpp

```diff
@@ -444,8 +444,9 @@ auto CheckUnit::CheckRequiredDefinitions() -> void {
                     "no definition found for declaration in impl file");
 
   // Note that more required definitions can be added during this loop.
-  for (size_t i = 0; i != context_.definitions_required().size(); ++i) {
-    SemIR::InstId decl_inst_id = context_.definitions_required()[i];
+  // NOLINTNEXTLINE (modernize-loop-convert)
```

Comment:
```suggestion
  // NOLINTNEXTLINE(modernize-loop-convert)
```

If there's a space between them, `(modernize-loop-convert)` gets interpreted as a free-text comment, and so this would silence all clang-tidy warnings. See [here](https://clang.llvm.org/extra/clang-tidy/#clang-tidy-nolint).

(Idle thought: making `definitions_required_by_decl` a queue and looping until it's empty might make this code clearer, and would avoid the need to suppress this warning. I don't know if it's worth the trouble, though.)